### DISCUSSION TITLE
Using magma with a project instead of as part of the project

### DISCUSSION THREAD

**Discussion by capitalist** [marcelotto/magma#5] [Ideas]
This way, you wouldn't need to install magma into the project to use it to build documentation. However, instead of `use Magma` to inject the docs, you need to actually modify the source file programmatically injecting the docs as if they were written into the module to begin with.

An added benefit, you could inject the function docs that get generated at that time as well.

The main drawback is that you can't continue to evolve your docs in Obsidian without a step to replace them in the source files.

**Comment by marcelotto** [comment]
Yes, I understand your concern about the current requirement for Magma to be bound within an Elixir project, which indeed limits its user base to Elixir developers. While I prefer to use Magma in my projects in precisely this way, I recognize that not all Elixir developers may prefer such an approach, and moreover, Magma has the potential to be useful in a broader context, not limited to documentation generation for code.

The strategy I would propose to accommodate both usage scenarios would be to introduce a CLI and distribute it as a self-contained binary (with something like Burrito). This would eliminate not only the need to install Magma directly into the project but also the requirement to have a complete Erlang/Elixir environment installed. However, I'm not sure when I will have time for this (see the [Roadmap](https://hexdocs.pm/magma/magma-user-guide-current-limitations-and-roadmap-article-section.html#roadmap)). Any help is much appreciated.
